src/test/java/org/candlepin/controller/ConsumerManagerTest.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2009 - 2024 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+
+package org.candlepin.controller;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.argThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import org.candlepin.model.Consumer;
+import org.candlepin.model.ConsumerCloudData;
+import org.candlepin.model.ConsumerCurator;
+import org.candlepin.service.EventAdapter;
+import org.candlepin.service.model.CloudCheckInEvent;
+import org.candlepin.test.TestUtil;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.Date;
+
+@ExtendWith(MockitoExtension.class)
+public class ConsumerManagerTest {
+
+    @Mock
+    private ConsumerCurator consumerCurator;
+
+    @Mock
+    private EventAdapter eventAdapter;
+
+    private ConsumerManager consumerManager;
+
+    @BeforeEach
+    void setUp() {
+        consumerManager = new ConsumerManager(consumerCurator, eventAdapter);
+    }
+
+    @Test
+    void testUpdateLastCheckInWithoutCloudData() {
+        Consumer consumer = mock(Consumer.class);
+
+        when(consumer.getConsumerCloudData()).thenReturn(null);
+
+        consumerManager.updateLastCheckIn(consumer);
+
+        verify(consumerCurator).updateLastCheckin(consumer);
+        verify(eventAdapter, never()).publish(any());
+    }
+
+    @Test
+    void testUpdateLastCheckInWithCloudData() {
+        Consumer consumer = mock(Consumer.class);
+        ConsumerCloudData consumerCloudData = mock(ConsumerCloudData.class);
```

Comment:
Same here with the mocks: data/model objects should absolutely never be mocked out

Reply:
ack done